test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -1127,14 +1127,12 @@ wrap f = first (MEither . bimap MErr f)
 -------------------------------------------------------------------------------}
 
 runIO ::
-     forall a h. (Class.IsTable h, NoThunks (Class.Session h IO))
+     forall a h. Class.IsTable h
   => LockstepAction (ModelState h) a
   -> LookUp (RealMonad h IO)
   -> RealMonad h IO (Realized (RealMonad h IO) a)
 runIO action lookUp = ReaderT $ \ !env -> do
-    x <- aux env action
-    assertNoThunks (envSession env) $ pure ()
-    pure x
+    aux env action
```

Comment:
So IIUC, this is where we were doing the check on each iteration, and now we're not, and doing it in the cleanup instead.

Reply:
Yes, and now instead of doing it for the `IO` version with both the mocked and real FS, we run the check only with the real FS.